File: apps/android/core/ai/src/main/java/com/scribblefit/core/ai/model/AnalysisSuggestion.kt
package com.scribblefit.core.ai.model

data class AnalysisSuggestion(
    val text: String,
    val emoji: String,
    val type: SuggestionType,
    val timestamp: Long
) {
    val fullText: String get() = "$text $emoji"
}

enum class SuggestionType {
    RECOVERY, PATTERN, MILESTONE, REST
}

File: apps/ios/ScribbleFit/ScribbleFit/Core/AI/AnalysisSuggestion.swift
import Foundation

public struct AnalysisSuggestion: Codable, Sendable {
    public let text: String
    public let emoji: String
    public let type: SuggestionType
    public let timestamp: Date

    public init(text: String, emoji: String, type: SuggestionType, timestamp: Date) {
        self.text = text
        self.emoji = emoji
        self.type = type
        self.timestamp = timestamp
    }

    public var fullText: String { "\(text) \(emoji)" }
}

public enum SuggestionType: String, Codable, Sendable {
    case recovery
    case pattern
    case milestone
    case rest
}
